packages/core/src/classes.ts

```diff
@@ -83,10 +83,14 @@ export interface UXObject extends Obj {
 /**
  * @public
  */
-export interface AttachedDoc extends Doc {
-  attachedTo: Ref<Doc>
-  attachedToClass: Ref<Class<Doc>>
-  collection: string
+export interface AttachedDoc<
+  Parent extends Doc = Doc,
+  Collection extends Extract<keyof Parent, string> | string = Extract<keyof Parent, string> | string,
```

Comment:
I think there can be some places where we don't create collection attributes for some reasons. Don't we have any problems with this now?

Reply:
It should be for any non system class. So let's try :-)

Reply:
@aonnikov 

> Don't we have any problems with this now?

We have, and it's not a problem, it's a designed behavior covered by tests. We will add a missing collection attribute even if it's not present on the model class and only added by mixin.

That's exactly why we have a trailing `| string` part here. The first leading part will let IDE suggest you the most appropriate options, while the second part will still let you type whatever you actually need.